### PR TITLE
docs: add libsinsp example code and documentation

### PR DESCRIPTION
**Pull request by dwindsor-scwx**
Signed-off-by: David Windsor <dwindsor@secureworks.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:
 /kind documentation

> Uncomment one (or more) `/area <>` lines:
/area libsinsp

**What this PR does / why we need it**:
add libsinsp example code and documentation

**Which issue(s) this PR fixes**: 
https://github.com/falcosecurity/libs/issues/31

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new: added libsinsp example code and documentation
```

### CODE REVIEW

**Review by leogr — COMMENTED** [review]
This is a great contribution! Thank you :100: 

Just one question: could we avoid depending on "chisel"? It would great if the example would not rely on that since we are trying to make "chisel" completely optional and not enabled by default.

**Comment by dwindsor-scwx** [comment]
> This is a great contribution! Thank you 💯
> 
> Just one question: could we avoid depending on "chisel"? It would great if the example would not rely on that since we are trying to make "chisel" completely optional and not enabled by default.

Sure! Does it make sense to perhaps split this one example into two example programs: one that demonstrates the usage of chisels/filters and one that doesn't? I'm not actually too clear on the separation between filters and chisels.

**Comment by leogr** [comment]
> > This is a great contribution! Thank you 100
> > Just one question: could we avoid depending on "chisel"? It would great if the example would not rely on that since we are trying to make "chisel" completely optional and not enabled by default.
> 
> Sure! Does it make sense to perhaps split this one example into two example programs: one that demonstrates the usage of chisels/filters and one that doesn't? I'm not actually too clear on the separation between filters and chisels.

A [chisel](https://github.com/draios/sysdig/wiki/Writing-a-Sysdig-Chisel,-a-Tutorial) is a Lua program that works on top filters and data collected by the libs. AFAIK, chisels are used only by `sysdig` CLI. Btw, they are completely optional. I recently moved them to the `userspace/chisel` folder for this reason, so consumers can avoid depending on them (and their dependencies) when they are not needed. Maybe, one day they could become a separate library.

I see in your example, `chisel.h` is imported only for this :point_down: :
https://github.com/falcosecurity/libs/pull/34/files#diff-7a7e437b41e679bace70e956e04d3e2a2a70b54c51baf7209b2cdfe0295efaf5R97

That seems to me not really needed (I have not tested the code, but I guess so). Moreover, chisels are not related to Falco rules files and that line won't load rules from the given directory. 

So, my advice is just to do not include chisels in this example. We could later, in another PR, show chisel in another example eventually, even though I have no clear idea how to do that atm.

Let me know if you need further info and feel free to DM me over slack if you need any help! :+1:

File: userspace/libsinsp/examples/test.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
+/*
+Copyright (C) 2021 SecureWorks Corp.
+
+This file is part of sysdig.
```
Comment (leogr) [inline userspace/libsinsp/examples/test.cpp:4]:
```suggestion
Copyright (C) 2021 The Falco Authors.
```

File: userspace/libsinsp/examples/util.cpp [hunk 2]
```diff
@@ -0,0 +1,332 @@
+/*
+Copyright (C) 2021 SecureWorks Corp.
+
+This file is part of sysdig.
```
Comment (leogr) [inline userspace/libsinsp/examples/util.cpp:4]:
```suggestion
Copyright (C) 2021 The Falco Authors.
```

File: userspace/libsinsp/examples/util.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (C) 2021 SecureWorks Corp.
+
+This file is part of sysdig.
```
Comment (leogr) [inline userspace/libsinsp/examples/util.h:4]:
```suggestion
Copyright (C) 2021 The Falco Authors.
```

**Review by leogr — CHANGES_REQUESTED** [review]
I've added some suggestions to set the correct license headers.
See below.

File: userspace/libsinsp/examples/test.cpp (hunk 1 above)
Comment (dwindsor) [reply userspace/libsinsp/examples/test.cpp]: Fixed

File: userspace/libsinsp/examples/util.cpp (hunk 2 above)
Comment (dwindsor) [reply userspace/libsinsp/examples/util.cpp]: Fixed

File: userspace/libsinsp/examples/util.h (hunk 3 above)
Comment (dwindsor) [reply userspace/libsinsp/examples/util.h]: Fixed

**Review by leogr — APPROVED** [review]

File: userspace/libsinsp/examples/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Example sinsp code
+
+This directory contains a program that demonstrates how to use libsinsp for event capture and filtering.
+
+## Quick Start ##
+
+`sinsp-example` monitors the host and any running containers for system activity. By default, it prints events of all types and is very noisy. 
+
+To use filtering, specify a [Sysdig filter](https://github.com/draios/sysdig/wiki/Sysdig-User-Guide#all-supported-filters) using `-f`.
```
Comment (leodido) [inline userspace/libsinsp/examples/README.md:9]:
```suggestion
To use filtering, specify a [filter](https://falco.org/docs/rules/supported-fields/#system-calls-source-syscall) using `-f`.
```

**Review by leodido — COMMENTED** [review]
Thanks for submitting this @dwindsor-scwx

I only left a suggestion (mind accepting it?) to link to falcosecurity docs

**Review by dwindsor — DISMISSED** [review]

**Comment by dwindsor-scwx** [comment]
> Thanks for submitting this @dwindsor-scwx
> 
> I only left a suggestion (mind accepting it?) to link to falcosecurity docs

I tried accepting it, but from the wrong github account accidentally.

**Review by leodido — APPROVED** [review]

**Review by leodido — DISMISSED** [review]

**Review by leogr — APPROVED** [review]
/approve

**Pull request merged**